controllers/resource/internal/test/util.go

```diff
@@ -0,0 +1,11 @@
+package util
```

Comment:
I think you can get away with duplicating this where its needed - seems to be test only. 

On a side note, package names should be the same as their directory names and `util` packages are strongly discouraged. Instead, prefer duplication until you can see a bundle of related utility functions and then consider creating a fosuced utility package like `secretutil`.

Reply:
This is really helpful, thanks 👍 Came to the same place after an offline conversation with @g-gaston .